code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,261 @@
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_19_invalid_access_token
+  Scenario: Invalid access token
+    Given the header "Authorization" is set to an invalid access token
+    And the request body is set to a valid request body
+    When the HTTP "POST" request is sent
+    Then the response status code is 401
+    And the response header "Content-Type" is "application/json"
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+     @geofencing_subscriptions_20_sink_credentials
+  Scenario:   Validate that the subscribed events are received in the sink, with the right sinkCredential, for those situations specified in the API.
+    Given the appropriate values are used for geofencing
+    When the  create subscription method is triggered  for subscription-id
+    Then the response  code is 201
+	  Then the get event details observed on notifications-url
+	  Then the subscribed event  received on notifications-url & sink credentials are as expected
+	  Then the response  code is 201
+	  And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    # The response has to comply with the generic response schema which is part of the spec
+    And the response body complies with the OAS schema at "/components/schemas/Subscription"
+
+
+
+   @geofencing_subscriptions_21_subscriptionExpireTime
+  Scenario:   For subscriptions that provide subscriptionExpireTime, validate that the subscribed events are not longer received after the expiration time.
+    Given the appropriate values are used for geofencing
+    When the  create subscription method is triggered  for subscription-id with subscriptionExpireTime after 5 seconds
+	  Then the subscriptionExpireTime value is reached
+	  Then the get event details observed on notifications-url
+	  Then the response  code is 404
+    And the response property "$.status" is 404
+    And the response property "$.code" is "NOT_FOUND"
+    And the response property "$.message" contains a user friendly text
+    
+
+  @geofencing_subscriptions_22_subscriptionMaxEvents
+  Scenario: For subscriptions that provide subscriptionMaxEvents, validate that the subscribed events are not longer received after the maximum events limit is reached.
+    Given the appropriate values are used for geofencing
+    When the  create subscription method is triggered  for subscription-id
+    Then the response  code is 201
+	  Then the  create subscription method is triggered  for subscription-id
+	  Then the subscriptionMaxEvents value is exceeded
+	  Then the get event details observed on notifications-url
```

Comment:
Idem to the scenario above. What doe this step mean? To test this scenario, you need an existing subscription with subscriptionMaxEvents , and with a pre-condition that the max events have already been triggered, when the device enters or leaves the geofence (depending on the subscription) no event is received in the callback